src/txn/txn_timestamp.c

```diff
@@ -107,27 +107,38 @@ __wt_txn_parse_timestamp(WT_SESSION_IMPL *session,
 
 #if WT_TIMESTAMP_SIZE == 8
 	{
-	static const u_char hextable[] = {
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  1,  2,  3,  4,  5,  6,  7,
-	    8,  9,  0,  0,  0,  0,  0,  0,
-	    0, 10, 11, 12, 13, 14, 15,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0,  0,  0,  0,  0,  0,  0,  0,
-	    0, 10, 11, 12, 13, 14, 15
+	static const char hextable[] = {
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
+	     0,  1,   2,   3,   4,   5,   6,   7,
+	     8,  9,  -1,  -1,  -1,  -1,  -1,  -1,
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
+	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
+	    -1, 10,  11,  12,  13,  14,  15, -1
 	};
 	wt_timestamp_t ts;
 	size_t len;
-	const char *hex;
-
-	for (ts.val = 0, hex = cval->str, len = cval->len; len > 0; --len)
-		ts.val = (ts.val << 4) | hextable[(int)*hex++];
+	int hex_val, hextable_count;
+	const char *hex_itr;
+
+	hextable_count = sizeof(hextable)/sizeof(hextable[0]);
```

Comment:
There is a macro for this: `WT_ELEMENTS(hextable)`.  There's no reason to put the count in a variable, as far as I can see: the compiler should figure out that it is a constant either way, but this just makes it work harder.